app-backend/db/src/main/scala/SceneToLayerDao.scala

```diff
@@ -39,54 +46,60 @@ object SceneToLayerDao
     FROM
   """ ++ tableF
 
+  def mosaicDefCacheKey(projectLayerId: UUID): String =
+    s"SceneToLayer:ProjectLayer:$projectLayerId:MultiTiff:${Config.publicData.enableMultiTiff}"
+
+  def deleteMosaicDefCache(projectLayerId: UUID): ConnectionIO[Unit] = {
+    for {
+      _ <- {
+        val cacheKey = mosaicDefCacheKey(projectLayerId)
+        logger.debug(s"Removing $cacheKey")
+        remove(cacheKey)(mosaicDefinitionCache, async[ConnectionIO]).attempt
+      }
+    } yield ()
+  }
+
   def acceptScene(projectLayerId: UUID, sceneId: UUID): ConnectionIO[Int] = {
-    fr"""
+    for {
+      acceptCount <- fr"""
       UPDATE scenes_to_layers
       SET accepted = true
       WHERE project_layer_id = ${projectLayerId}
         AND scene_id = ${sceneId}
-    """.update.run
+        """.update.run
+      _ <- deleteMosaicDefCache(projectLayerId)
+    } yield acceptCount
   }
 
   def acceptScenes(
       projectLayerId: UUID,
       sceneIds: List[UUID]
   ): ConnectionIO[Int] = {
-    sceneIds.toNel match {
-      case Some(ids) => acceptScenes(projectLayerId, ids)
-      case _         => 0.pure[ConnectionIO]
-    }
+    for {
+      updateCount <- sceneIds.toNel match {
+        case Some(ids) => acceptScenes(projectLayerId, ids)
+        case _         => 0.pure[ConnectionIO]
+      }
+      _ <- deleteMosaicDefCache(projectLayerId)
```

Comment:
no reason to delete mosaic defs if there were no sceneIds

Reply:
Good point

Reply:
Addressed in https://github.com/raster-foundry/raster-foundry/pull/5192/commits/7045116648b7299562b65c913cb5b738963bb2fa